src/main/java/gregtech/api/render/MetaTileEntityTESR.java

```diff
@@ -125,10 +125,21 @@ public void renderTileEntityFast(MetaTileEntityHolder te, double x, double y, do
 
     @Override
     public boolean isGlobalRenderer(MetaTileEntityHolder te) {
-        if(te instanceof IRenderMetaTileEntity) {
-            return ((IRenderMetaTileEntity) te).isGlobalRenderer();
-        } else if(te instanceof IFastRenderMetaTileEntity) {
-            return ((IFastRenderMetaTileEntity) te).isGlobalRenderer();
+        if (te instanceof IRenderMetaTileEntity) {
+            if (((IRenderMetaTileEntity) te).isGlobalRenderer()) {
+                return true;
+            }
+        } else if (te instanceof IFastRenderMetaTileEntity) {
+            if (((IFastRenderMetaTileEntity) te).isGlobalRenderer()) {
+                return true;
+            }
+        }
+        if (te.getMetaTileEntity() instanceof IRenderMetaTileEntity) {
+            if (((IRenderMetaTileEntity) te.getMetaTileEntity()).isGlobalRenderer()) {
+                return true;
+            }
+        } else if (te.getMetaTileEntity() instanceof IFastRenderMetaTileEntity){
+            return ((IFastRenderMetaTileEntity) te.getMetaTileEntity()).isGlobalRenderer();
```

Comment:
It looks still way too complicated to me. I would write it like this:
```suggestion
        if (te instanceof IRenderMetaTileEntity) {
			return ((IRenderMetaTileEntity) te).isGlobalRenderer();
        } else if (te instanceof IFastRenderMetaTileEntity) {
			return ((IFastRenderMetaTileEntity) te).isGlobalRenderer();
        }
		
        if (te.getMetaTileEntity() instanceof IRenderMetaTileEntity) {
			return ((IRenderMetaTileEntity) te.getMetaTileEntity()).isGlobalRenderer();
        } else if (te.getMetaTileEntity() instanceof IFastRenderMetaTileEntity){
            return ((IFastRenderMetaTileEntity) te.getMetaTileEntity()).isGlobalRenderer();
```
But my code is invalid if `te` can be `I*RenderMetaTileEntity` and `MetaTileEntity` it contains can be `I*RenderMetaTileEntity` at same time. So can it?

Reply:
This change also work well at present, as far as I know no `MetatileEntityHolder` implements these two interfaces now, so it's still essentially a judgment for `MetatileEntity`. However,  `te` and `MetatileEntity` should be cared at same time. Logically, we should ensure that it returns true whenever one of `MetatileEntityHolder` or `MetatileEntity` it contains has a `GlobalRenderer` requirement. In addition, returning true is still work fine for `MetatileEntityHolder` even if `GlobalRenderer` is not required.

Reply:
I understand let's leave current implementation.